dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/analytics/dimension/mappers/ProgramStageDataElementMapperTest.java

```diff
@@ -57,7 +57,7 @@ void testProgramStageDataElementMapper()
             ImmutableList.of( this::setDataElement, this::setProgramStage ),
             ImmutableList.of(
                 Pair.of( DimensionResponse::getValueType, VALUE_TYPE ),
-                Pair.of( DimensionResponse::getId, PROGRAM_STAGE_UID + "." + DATA_ELEMENT_UID ) ) );
+                Pair.of( DimensionResponse::getId, DATA_ELEMENT_UID ) ) );
```

Comment:
it's because the logic for TEI is different than before.
Basically the mapper now always add prefix if present, while before it was added depending on the type of item to map.
To keep the behaviour consistent with before, I introduced "PrefixStrategy", which will map keeping the old behaviour for events and enrollments